partiql-lib/partiql-sprout/src/main/kotlin/org/partiql/sprout/generator/target/kotlin/spec/KotlinNodeSpec.kt

```diff
@@ -30,25 +30,33 @@ import org.partiql.sprout.model.TypeDef
  * @property builder        Type builder
  * @property constructor    Implementation constructor
  * @property companion      A place for static methods on nodes
- * @property types          Non-node types defined within this node
+ * @property nodes          Node types defined within this node
  */
 sealed class KotlinNodeSpec(
     val def: TypeDef,
     val clazz: ClassName,
     val builder: TypeSpec.Builder,
     val constructor: FunSpec.Builder,
     val companion: TypeSpec.Builder,
-    val types: MutableList<TypeSpec> = mutableListOf(),
+    val ext: MutableList<TypeSpec> = mutableListOf(),
 ) {
 
-    open val children: List<KotlinNodeSpec> = emptyList()
+    /**
+     * Nodes defined within this node, but don't inherit from this class
+     */
+    abstract val nodes: List<KotlinNodeSpec>
+
+    /**
+     * All types defined within this node
+     */
+    abstract val children: List<KotlinNodeSpec>
 
     /**
      * Returns the built Pair<Base, Impl>
      */
     open fun build(): TypeSpec = with(builder) {
         primaryConstructor(constructor.build())
-        types.forEach { addType(it) }
+        ext.forEach { addType(it) }
```

Comment:
Tracking enums.

Reply:
ext is an extension point to add any KotlinPoet TypeSpec to a node. For the built in poems, this just means enums.